assets/report_template.Rmd

```diff
@@ -894,7 +892,7 @@ invisible(dev.off())
 cat("## DADA2\n")
 
 # indicate reference taxonomy
-if (!params$flag_dada_ref_tax_user) {
+if ( !isFALSE(params$dada2_ref_tax_title) ) {
```

Comment:
This looks odd to me -- negating `isFALSE` instead of testing for true -- but maybe it's the way it should be?
(In (at least) one more place further down.)

Reply:
True that looks odd. I actually havent tested `isTRUE` but used `!isFALSE` because the default of that params is `FALSE`, if data is coming through it could be anything, TRUE, a string (file, setting, ...), a number,...
So I figured `!isFALSE` is my safest bet. Am I wrong?

Reply:
I've never used either of the two functions, but `isTRUE` would be my choice even if the default for the param is false. The difference between `if ( isTRUE(cond) )` and `if ( cond )` appears to be that `isTRUE` doesn't return NAs. `isFALSE` is defined analogously according to the documentation. `isTRUE('string')` returns false while evaluating a string as a logical returns NA. If you're not certain of what type the variable is (e.g. a logical or a string) I think you need to do something more elaborate.

Reply:
Yes, the point is here that I'm not certain of what type the variable is (e.g. a logical or a string) and 
```
!isFALSE('string') 
!isFALSE(TRUE)
!isFALSE(1)
```
all return TRUE, so seem to work as intended.
`if ( 'string' ) {} `returns `Error in if ("string") { : argument is not interpretable as logical`
`isTRUE('string')` is `FALSE`, so not doing what I want.
So while I agree the double negation looks bad, it seems to do what it should and the cleaner versions I came up with didnt.

Reply:
I am open to suggestion ofc ;)

Reply:
I suppose the correct thing would be to make sure the variable is logical before using it. I understand if you have better things to do than correct working code though.